npm/angular-signals/src/mount.ts

```diff
@@ -0,0 +1,536 @@
+}
+
+// currently not a great way to detect if a function is a Writable Signal. Need to improve this.
+function isWritableSignal (prop: any): boolean {
+  return isSignal(prop) && typeof prop === 'function' && typeof prop.set === 'function'
+}
+
+function convertPropertyToSignalIfApplicable (propValue: any, componentValue: any, injector: Injector) {
+  const isComponentValueAnInputSignal = isInputSignal(componentValue)
+  const isComponentValueAModelSignal = isModelSignal(componentValue)
+  let convertedValueIfApplicable = propValue
+
+  // If the component has the property defined as an InputSignal, we need to detect whether a non signal value or not was passed into the component as a prop
+  // and attempt to merge the value in correctly.
+  // We don't want to expose the primitive created signal as it should really be one-way binding from within the component.
+  // However, to make CT testing easier, a user can technically pass in a signal to an input component and assert on the signal itself and pass in updates
+  // down to the component as 1 way binding is supported by the test harness
+  if (isComponentValueAnInputSignal) {
+    const isPassedInValueNotASignal = !isSignal(propValue)
+
+    if (isPassedInValueNotASignal) {
+      // Input signals require an injection context to set initial values.
+      // Because of this, we cannot create them outside the scope of the component.
+      // Options for input signals also don't allow the passing of an injection contexts, so in order to work around this,
+      // we convert the non signal input passed into the input to a writable signal
+      convertedValueIfApplicable = signal(propValue)
+    }
+
+    // If the component has the property defined as a ModelSignal, we need to detect whether a signal value or not was passed into the component as a prop.
+    // If a non signal property is passed into the component model (primitive, object, array, etc), we need to set the model to that value and propagate changes of that model through the output spy.
+    // Since the non signal type likely lives outside the context of Angular, the non signal type will NOT be updated outside of this context. Instead, the output spy will allow you
+    // to see this change.
+    // If the value passed into the property is in fact a signal, we need to set up two-way binding between the signals to make sure changes from one propagate to the other.
+  } else if (isComponentValueAModelSignal) {
+    const isPassedInValueLikelyARegularSignal = isWritableSignal(propValue)
+
+    // if the value passed into the component is a signal, set up two-way binding
+    if (isPassedInValueLikelyARegularSignal) {
+      // update the passed in value with the models updates
+      componentValue.subscribe((value: any) => {
+        propValue.set(value)
+      })
+
+      // update the model signal with the properties updates
+      toObservable(propValue, {
+        injector,
+      }).subscribe((value) => {
```

Comment:
This was a memory leak. The observable by the looks of it hangs around because the subscription is active. To fix this, I pushed the subscription into a list to clean up when we clean up the fixture. Updated and fixed in 12c7e36238. Really good catch!